File: majorProject/src/main/java/com/aocc/majorproject/ui/UiBanner.java
package com.aocc.majorproject.ui;

import android.graphics.Color;
import android.graphics.Paint;

import com.aocc.framework.Graphics;

/** Large centred prompt or heading text (e.g. "Game Over!", "Press anywhere to start"). */
public class UiBanner {

    private final float textSize;
    private final int color;

    public UiBanner(float textSize) {
        this(textSize, Color.WHITE);
    }

    public UiBanner(float textSize, int color) {
        this.textSize = textSize;
        this.color = color;
    }

    public void paint(Graphics g, Paint paint, String text, int centerX, int centerY) {
        float previousSize = paint.getTextSize();
        paint.setTextSize(textSize);
        UiText.drawCentered(g, paint, text, centerX, centerY, color);
        paint.setTextSize(previousSize);
    }
}

File: majorProject/src/main/kotlin/com/aocc/majorproject/ui/UiBanner.kt
package com.aocc.majorproject.ui

import android.graphics.Color
import android.graphics.Paint
import com.aocc.framework.Graphics

/** Large centred prompt or heading text (e.g. "Game Over!", "Press anywhere to start"). */
class UiBanner(
    private val textSize: Float,
    private val color: Int = Color.WHITE,
) {

    constructor(textSize: Float) : this(textSize, Color.WHITE)

    fun paint(g: Graphics, paint: Paint, text: String, centerX: Int, centerY: Int) {
        val previousSize = paint.textSize
        paint.textSize = textSize
        UiText.drawCentered(g, paint, text, centerX, centerY, color)
        paint.textSize = previousSize
    }
}
